### PR TITLE
ci: Add Rust Windows cross toolchain on ubuntu-rolling image

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

**Comment by xclaesse** [comment]
CI image updates must be merged before actually using it, otherwise CI still uses the old image. See https://github.com/mesonbuild/meson/pull/11990.

File: ci/ciimage/ubuntu-rolling/install.sh [hunk 1]
```diff
@@ -50,6 +51,15 @@ dub_fetch dubtestproject
 dub build dubtestproject:test1 --compiler=ldc2
 dub build dubtestproject:test2 --compiler=ldc2
 
+# Remove debian version of Rust and install latest with rustup.
+# This is needed to get the cross toolchain as well.
+apt-get -y remove rustc || true
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup-init.sh
+chmod +x rustup-init.sh
+./rustup-init.sh -y
```
Comment (tristan957) [inline ci/ciimage/ubuntu-rolling/install.sh:59]: Might be worth doing something like: https://git.sr.ht/~tristan957/dotfiles/tree/master/item/bootstrap/shared/rustup.sh#L2

**Comment by tristan957** [comment]
CI failures don't make sense to me

File: ci/ciimage/ubuntu-rolling/install.sh (hunk 1 above)
Comment (xclaesse) [reply ci/ciimage/ubuntu-rolling/install.sh]: Can be simplified with `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y` indeed.

**Comment by xclaesse** [comment]
> CI failures don't make sense to me

Building docker images has been failing for a while, @eli-schwartz fixed ubuntu-rolling recently which is the only one we care here. Fixing others would be great but unrelated to this PR.

**Comment by xclaesse** [comment]
ubuntu-rolling did pass Rust tests successfully, so I think this PR is ready: https://github.com/mesonbuild/meson/actions/runs/5611071724/jobs/10266902617?pr=12015#step:5:1792.

**Comment by tristan957** [comment]
Cool. I say go ahead and merge it.

File: ci/ciimage/ubuntu-rolling/install.sh [hunk 2]
```diff
@@ -27,6 +27,7 @@ pkgs=(
   bindgen
   itstool
   openjdk-11-jre
+  curl
```
Comment (eli-schwartz) [inline ci/ciimage/ubuntu-rolling/install.sh:30]: Shouldn't that image have wget instead?

Comment (xclaesse) [reply ci/ciimage/ubuntu-rolling/install.sh]: Indeed, I was using the command from https://rustup.rs/, but changed to wget now, that probably reduce a bit the docker image size.

**Comment by xclaesse** [comment]
> In order for the tests to still pass on master, we just need a drop-in compatible replacement rustc

While working on this I noticed that's not exactly true, if rustc is missing those tests are happily skipped.

**Comment by eli-schwartz** [comment]
Right, that's true... if the entire category is skipped because the tool isn't found at all then a single test skipping isn't considered an "unexpected skip" error.

**Comment by eli-schwartz** [comment]
```
info: default host triple is x86_64-unknown-linux-gnu
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: latest update on 2023-07-13, rust version 1.71.0 (8ede3aae2 2023-07-12)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'
```

I don't think we need the docs or rustfmt, is it possible to exclude those?

**Comment by tristan957** [comment]
I think you want to add `--profile minimal` to the shell script invocation.

https://rust-lang.github.io/rustup/concepts/profiles.html

**Comment by eli-schwartz** [comment]
`--profile minimal --component clippy` I guess? We actually do have `test_rust_clippy` so that actually accomplishes something. On the other hand, we test this on other CI machines anyway.

**Comment by xclaesse** [comment]
> `--profile minimal --component clippy` I guess?

That worked locally, updated the PR to see how it goes.

**Comment by xclaesse** [comment]
It worked, let's merge so I can use it in the other PR.

**Pull request merged**